### PR TITLE
needs-restarting: add --reboothint option

### PR DESCRIPTION
**Pull request by dmnks**
This is the original implementation from the needs-restarting script in
yum-utils, with a few tweaks (e.g. no versions are printed to keep
things simple).

Original RHEL-7 request:
https://bugzilla.redhat.com/show_bug.cgi?id=1192946

New RHEL-8 request:
https://bugzilla.redhat.com/show_bug.cgi?id=1639468

### CODE REVIEW

File: plugins/needs_restarting.py [hunk 1]
```diff
@@ -35,6 +35,11 @@
 import stat
 
 
+# For which package updates we should recommend a reboot
+# Taken from https://access.redhat.com/solutions/27943
+NEED_REBOOT = ['kernel', 'glibc', 'linux-firmware', 'systemd', 'dbus']
```
Comment (Conan-Kudo) [inline plugins/needs_restarting.py]: `dbus-broker` and `dbus-daemon` are also probably ones that belong on this list...

File: plugins/needs_restarting.py [hunk 2]
```diff
@@ -180,6 +188,27 @@ def run(self):
...
 
+        if self.opts.reboothint:
+            need_reboot = set()
+            installed = self.base.sack.query().installed()
+            for pkg in installed.filter(name=NEED_REBOOT):
+                if pkg.installtime > process_start.boot_time:
+                    need_reboot.add(pkg.name)
+            if need_reboot:
+                print('Core libraries or services have been updated '
+                      'since boot-up:')
```
Comment (m-blaha) [inline plugins/needs_restarting.py]: All these static strings should be marked for translation using _()

Comment (dmnks) [reply plugins/needs_restarting.py]: Good point, thanks

File: plugins/needs_restarting.py (hunk 1 above)
Comment (dmnks) [reply plugins/needs_restarting.py]: Those are not mentioned explicitly on the KB page (only "dbus" is), but yeah, they seem to be quite essential so I'll include them. In any case, we should think about a more flexible way in the future as having a hard-coded list doesn't scale well :)

Comment (dmnks) [reply plugins/needs_restarting.py]: Also, I'd like to revisit this plugin later as it deserves more love and could be more useful and user-friendly. But that's for later.

**Comment by Conan-Kudo** [comment]
@dmnks A more flexible solution might be something like what `urpmi` does: it reads for packages in the transaction that contain the Provides statement: `should-restart = system`. That lets it scale a lot better.

**Comment by dmnks** [comment]
That's really nice, I like that.

The only downside I can see is that this would require an action from packagers. However, there's just a handful of them at the moment so it shouldn't be too hard. And it's also their domain of expertise (whether to reboot or not), after all :)

It's also worth mentioning that RH errata have been shipping (for a few years) with a special flag (in the updateinfo data) with a similar meaning, i.e. whether a reboot is required. Neither yum or dnf have been adapted to read that flag so far. I can see the benefit here, though: sometimes, you may only need reboots after a specific set of packages is updated (as opposed to one specific package). You can't model that with the simple Provides flag.

Nevertheless, a lot of interesting thoughts here. Let's see where we can take this in the near future.

**Comment by Conan-Kudo** [comment]
@dmnks It may be worth it to support both models. The issue with updateinfo is that it's only read when updateinfo is specifically requested (that is, as part of an advisory request in dnf).

**Comment by dmnks** [comment]
Right. I agree; both ways should be combined somehow (perhaps accessible via a single API query or something).

File: plugins/needs_restarting.py (hunk 1 above)
Comment (dmnks) [reply plugins/needs_restarting.py]: Note to self: the plugin could also be renamed to "ps" for simplicity (and consistency with zypper).

**Pull request merged**